### PR TITLE
Extract Components from Step.html

### PR DESCRIPTION
**Pull request by eggerschwiler**
Extracted the following components from Step.html/StepController:
- FallbackComponent (to display the fallback message)
- NavigationToolbarComponent (the step/page navigation toolbar)
- ComparisonViewComponent (the whole content of the comparison tab)
- ComparisonToolbarComponent (the toolbar with the buttons to change the comparison view)

@danielsuter could you please review these changes, so that we can merge them into develop?

### CODE REVIEW

File: scenarioo-client/app/step/comparison/comparisonToolbar.component.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+            step: '<',
+            comparisonViewOptions: '<',
+        },
+        template: require('./comparisonToolbar.html'),
+        controller: ComparisonToolbarController,
+    });
+function ComparisonToolbarController(localStorageService, ConfigurationService: ConfigurationService) {
+    const ctrl = this;
+
+    ctrl.setComparisonView = (viewId) => {
```
Comment (danielsuter) [inline scenarioo-client/app/step/comparison/comparisonToolbar.component.ts]: add typing information, string?

File: scenarioo-client/app/step/comparison/comparisonToolbar.component.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+    ctrl.switchComparisonSingleScreenView = () => {
+        const viewId = ctrl.isComparisonView('CurrentScreen') ? 'OtherScreen' : 'CurrentScreen';
+        ctrl.setComparisonView(viewId);
+    };
+
+    ctrl.isComparisonChangesToBeHighlightedAvailable = () => ctrl.step && ctrl.step.diffInfo && ctrl.step.diffInfo.changeRate !== 0 && !ctrl.step.diffInfo.isAdded;
+
+    ctrl.isComparisonChangesHighlighted = () => {
+        // highlighting is turned on, and there are changes in this screenshot to be highlighted
```
Comment (danielsuter) [inline scenarioo-client/app/step/comparison/comparisonToolbar.component.ts]: no need to use {} and return statement

File: scenarioo-client/app/step/comparison/comparisonToolbar.component.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    };
+
+    ctrl.toggleComparisonChangesHighlighted = () => {
+        ctrl.comparisonViewOptions.changesHighlighted = !ctrl.comparisonViewOptions.changesHighlighted;
+        setLocalStorageValue('diffViewerStepComparisonChangesHighlighted', ctrl.comparisonViewOptions.changesHighlighted);
+    };
+
+    ctrl.getComparisonViewHighlightChangesColor = () => ConfigurationService.diffViewerDiffImageColor();
+
+    function setLocalStorageValue(storageKey, value) {
```
Comment (danielsuter) [inline scenarioo-client/app/step/comparison/comparisonToolbar.component.ts]: typing

File: scenarioo-client/app/step/index.ts [hunk 4]
```diff
@@ -5,6 +5,10 @@ import './screenAnnotations/screenAnnotationsButton.directive';
 import './screenAnnotations/screenAnnotationInfoPopup.controller';
 import './screenAnnotations/screenAnnotations.service';
 import './stepNotFound/stepNotFound.component';
+import './fallback/fallback.component';
+import './navigation/navigation.component';
```
Comment (danielsuter) [inline scenarioo-client/app/step/index.ts]: navigation might be duplicated. What about StepNavigation?

Comment (eggerschwiler) [reply scenarioo-client/app/step/index.ts]: Good point. I renamed the component and html to more clearly differentiate it from the shared Naviation.

**Pull request merged**